src/requestManager.cpp

```diff
@@ -286,7 +286,7 @@ CNodeRequestData::CNodeRequestData(CNode* n)
   desirability -= latency;
 }
 
-void CUnknownObj::AddSource(CNode* from)
+bool CUnknownObj::AddSource(CNode* from)
 {
   if (std::find_if(availableFrom.begin(), availableFrom.end(), IsCNodeRequestDataThisNode(from)) == availableFrom.end())  // node is not in the request list
```

Comment:
why not prefer the form `begin(availableFrom), end(availableFrom)`?
brace alignment needs fixing here too.